Added Wishlist functionality

dominiqueroux opened the pull request:
Same functionality as shopping cart:
- User can add products to wishlist
- User can remove products from wishlist
- User can update entries from wishlist

Differences:
- User cannot buy directly a wishlist
- User can move all wishlist entries to the shopping cart

dominiqueroux committed:
Iplementation of the wishlist

It has the same functionalities as the Shopping Cart.
Only difference, you cannot buy a wishlist.

Tests are adapted as well.

dominiqueroux committed:
Updated the wishlist functionality:
User is now able to move all products from the wishlist to the shopping cart

KnownInter reviewed (changes requested):
- The user should be able to add the products to their wishlist regardless of the status
- `amountOrTime` is unnecessary in a wishlist
- `update` function is unnecessary since we won't use `amountOrTime`
- `move` function should be specific to a product and should not apply to the whole wishlist (also check if product is available if you implement point 1)

dominiqueroux commented:
> * The user should be able to add the products to their wishlist regardless of the status

Agree on that

> * `amountOrTime` is unnecessary in a wishlist
> * `update` function is unnecessary since we won't use `amountOrTime`
> * `move` function should be specific to a product and should not apply to the whole wishlist (also check if product is available if you implement point 1)

If we don't have the `amountOrTime` in the wishlist, then we cannot add the product directly to the shopping cart, since this is needed.

dominiqueroux committed:
Updated Wishlist:
- User simply adds a product (no amountOrTime)
- User can therefore, only delete a product from the wishlist and not update

KnownInter committed:
Removed some unnecessary code
